packages/front-end/services/experiments.ts

```diff
@@ -245,6 +245,25 @@ export function applyMetricOverrides(
   return { newMetric, overrideFields };
 }
 
+export function isExpectedDirection(
+  stats: SnapshotMetric,
+  metric: MetricInterface
+): boolean {
+  const expected: number = stats?.expected ?? 0;
+  return metric.inverse ? expected < 0 : expected > 0;
+}
+
+export function isStatSig(
+  stats: SnapshotMetric,
+  pValueThreshold: number
+): boolean {
+  const pValue: number = stats?.pValue ?? 1;
+  return pValue < pValueThreshold;
+}
+
 export function pValueFormatter(pValue: number) {
```

Comment:
it looks like the few places where this function is called may have the possibility of either erroring or being undefined. Specifically:
pValueGuardrailResults.tsx: `pValueFormatter(r.stats.pValue)`
pValueColumn.tsx: `pValueFormatter(stats.pValue)`

1. It's possible that stats could be undefined, so perhaps use optional chaining.
2. The function itself should probably allow passing undefined, like
`export function pValueFormatter(pValue?: number)`

Reply:
Alternatively we could check if the pValue is a number and only try to format a valid numeric pValue instead of handling all possible invalid input in this function, e.g.:

```ts
// React component implementation that needs to display the P-value text to the user

let pValueText = 'P-value is missing';
if (typeof r.stats?.pValue === 'number') {
  pValueText = pValueFormatter(r.stats.pValue)
}
```

It's arguably not the concern of a util to provide an empty state for invalid data passed in, and it's possible different implementations of this may want to provide different empty state text.

Also if we ever get to localizing the app and supporting other languages, UI strings should remain in the UI components and not be in the utils.

Reply:
@bryce-fitzsimons in both cases there is logic elsewhere that means it "shouldn't" ever be undefined. But I could add the optional chaining to be safe, I guess.

I agree with Tina that this would best be handled as a front end component and not in a util.